services/MerchStoreService.ts

```diff
@@ -72,11 +72,18 @@ export default class MerchStoreService {
       const currentCollection = await merchCollectionRepository.findByUuid(uuid);
       if (!currentCollection) throw new NotFoundError('Collection not found');
       let updatedCollection = await merchCollectionRepository.upsertMerchCollection(currentCollection, changes);
-      if (changes.discountPercentage) {
+      if (changes.discountPercentage !== undefined) {
         const { discountPercentage } = changes;
         await Repositories
           .merchStoreItemOption(txn)
           .updateMerchItemOptionsInCollection(uuid, discountPercentage);
+      }
+      if (changes.archived !== undefined) {
+        await Repositories
+          .merchStoreItem(txn)
+          .updateMerchItemInCollection(uuid, true);
+      }
+      if (changes.discountPercentage !== undefined || changes.archived !== undefined) {
```

Comment:
If an admin wants to un-archive a collection, do we still want to set all items to be hidden?